test/unit/test_environment.py

```diff
@@ -68,6 +69,17 @@ def test_read_key_serialized_hyperparameters():
     assert _env.read_hyperparameters() == ALL_HYPERPARAMETERS
 
 
+def test_read_key_serialized_and_non_key_serialized_hyperparameters():
+    hps_to_serialize = ('default_user_module_name', 'sagemaker_job_name', 'sagemaker-training-job',
```

Comment:
is `'sagemaker-training-job'` serving any purpose here?

Reply:
The test asserts that parsing the hyperparameters file work when some hyperparameters are JSON serialized and some are not.

So, these keys could any keys in the list of hyperparameters.

Reply:
but `'sagemaker-training-job'` is just a value in `SAGEMAKER_HYPERPARAMETERS` - I don't see it being used as a key anywhere

Reply:
```ALL_HYPERPARAMETERS == SAGEMAKER_HYPERPARAMETERS + USER_HYPERPARAMETERS```:

https://github.com/mvsusp/sagemaker-containers/blob/6e5ca1539d49e48888094b33ad4f085f66a6423b/test/unit/test_environment.py#L56

The test can less complicated. I changed the testes to make the test setup clear for better readability.